modules/router-store/src/provide_router_store.ts

```diff
@@ -0,0 +1,53 @@
+import { Provider } from '@angular/core';
+import {
+  _createRouterConfig,
+  _ROUTER_CONFIG,
+  ROUTER_CONFIG,
+  RouterState,
+  StoreRouterConfig,
+} from './router_store_config';
+import {
+  FullRouterStateSerializer,
+  SerializedRouterStateSnapshot,
+} from './serializers/full_serializer';
+import { MinimalRouterStateSerializer } from './serializers/minimal_serializer';
+import {
+  BaseRouterStoreState,
+  RouterStateSerializer,
+} from './serializers/base';
+import { _initRouterStore } from './init_router_store';
+
+/**
+ * Provider function to connect the RouterModule to the StoreModule
+ *
+ * Usage:
+ *
+ * ```typescript
+ * bootstrapApplication(AppComponent, {
+ *  providers: [
+ *    provideRouterStore()
+ *  ]
+ * })
+ * ```
+ */
+export function provideRouterStore<
+  T extends BaseRouterStoreState = SerializedRouterStateSnapshot
+>(config: StoreRouterConfig<T> = {}): Provider[] {
+  return [
+    { provide: _ROUTER_CONFIG, useValue: config },
+    {
+      provide: ROUTER_CONFIG,
+      useFactory: _createRouterConfig,
+      deps: [_ROUTER_CONFIG],
+    },
+    {
+      provide: RouterStateSerializer,
+      useClass: config.serializer
+        ? config.serializer
+        : config.routerState === RouterState.Full
+        ? FullRouterStateSerializer
+        : MinimalRouterStateSerializer,
+    },
+    _initRouterStore,
+  ];
+}
```

Comment:
We decided to use `EnvironmentProviders` as the return type of standalone functions to throw a compilation error if they're used in the wrong place (e.g. `Component/Directive.providers`).

```suggestion
export function provideRouterStore<
  T extends BaseRouterStoreState = SerializedRouterStateSnapshot
  // import `EnvironmentProviders` from `@ngrx/store` package
>(config: StoreRouterConfig<T> = {}): EnvironmentProviders {
  return {
    ɵproviders: [
      { provide: _ROUTER_CONFIG, useValue: config },
      {
        provide: ROUTER_CONFIG,
        useFactory: _createRouterConfig,
        deps: [_ROUTER_CONFIG],
      },
      {
        provide: RouterStateSerializer,
        useClass: config.serializer
          ? config.serializer
          : config.routerState === RouterState.Full
          ? FullRouterStateSerializer
          : MinimalRouterStateSerializer,
      },
      StoreRouterConnectingService,
      {
        provide: ENVIRONMENT_INITIALIZER,
        multi: true,
        useFactory() {
          return () => inject(StoreRouterConnectingService);
        },
      },
    ],
  };
}
```